### PR TITLE
feat(dropdown/combo-box): syncing dropdown and combo-box to v11

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
[#10103](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10103)
[#10106](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10106)

### Description
This PR syncs the Dropdown and ComboBox components in our CWC library to the Carbon React V11 library.

Some of the new features include the ability to display warning and invalid messages and icons, different menu directions, making the components readOnly, among other features.

One of the biggest changes to the ComboBox is that menu activation/opening can now only be done by clicking on the caret icon instead of the input element. 

### Changelog

**New**

- warning and invalid icons/messages
- menu directions (top/bottom)
- readOnly feature

**Changed**

- certain styles and class names had to be changed due to v11's revamp
- many of the component properties/attributes were renamed to fit v11's naming conventions
- ComboBox activation behavior can now only be activated through clicking the caret button

**Removed**

- some unneeded styles and props

### CODE REVIEW

File: packages/carbon-web-components/src/components/combo-box/combo-box-story.mdx
Comment (kennylam) [inline packages/carbon-web-components/src/components/combo-box/combo-box-story.mdx:1]: Maybe we can wrap this and `dropdown` stories in a div with an inline width like the React version so it's not 100% of the viewport (React version is `400px`).

**Review by kennylam — COMMENTED** [review]
@IgnacioBecerra This looks great. Just need to make sure all `BX` prefixes are changed to `CDS`.

For `combo-box`, the React version does expand when you click the input in the default story but not the playground. We might need to check with them to see which is correct. One other thing I noticed is when the component is `readonly` and has a selection, it's still possible to clear the selection with the `x` button. But it looks like that should be greyed out as well.

File: packages/carbon-web-components/src/components/combo-box/combo-box-story.mdx [hunk 1]
```diff
@@ -35,8 +29,8 @@ import '@carbon/web-components/es/components/combo-box/index.js';
 ```html
 <cds-combo-box
   helper-text="Optional helper text"
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/combo-box/combo-box-story.mdx:31]:
```suggestion
  helper-text="This is some helper text"
```

**Comment by annawen1** [comment]
Have some references that need to be cleaned up

Screenshot 2023-04-18 at 4 35 21 PM

File: packages/carbon-web-components/tests/spec/dropdown_spec.ts [hunk 2]
```diff
@@ -11,8 +11,8 @@ import { delay } from 'bluebird';
 import { render } from 'lit';
 import EventManager from '../utils/event-manager';
 
-import BXDropdown from '../../src/components/dropdown/dropdown';
-import BXDropdownItem from '../../src/components/dropdown/dropdown-item';
+import CDSDropdown from '../../src/components/dropdown/dropdown';
+import CDSDropdownItem from '../../src/components/dropdown/dropdown-item';
 import { Default } from '../../src/components/dropdown/dropdown-story';
```
[the remark this answers is not in the document]
Comment (annawen1) [reply packages/carbon-web-components/tests/spec/dropdown_spec.ts:16]: Need to reference the `Playground` story in the template as well

```
const template = (props?) =>
  Playground({
    'cds-dropdown': props,
  });
  ```
  
  I see we need to do this for the `combo-box_spec.ts` file too. For `slider__spec.ts` - I think I corrected that in a recently merged PR
  
  
Screenshot 2023-04-18 at 4 38 48 PM

**Review by annawen1 — COMMENTED** [review]
Looks good, just got some stuff to clean up

**Comment by IgnacioBecerra** [comment]
looking into the rest of the errors..

**Comment by IgnacioBecerra** [comment]
@kennylam Asked the Carbon team about the ComboBox Playground behavior (where the menu didn't open upon click), and turns out it was a bug after all. I changed it back to its original behavior 👍

**Review by kennylam — APPROVED** [review]
Thanks @IgnacioBecerra!

**Review by ariellalgilmore — APPROVED** [review]
LGTM! 💯

**Comment by RichKummer** [comment]
Hey @IgnacioBecerra , this looks really good! 

The only issue I found is that when the `dropdown` is set to `inline` AND has `warn` active, the icon appears over the helper text. I also don't see the warning text when set to `inline`

**Current build**
Screen Shot 2023-04-21 at 3 57 04 PM


**React**
Screen Shot 2023-04-21 at 3 59 30 PM

**Comment by RichKummer** [comment]
@IgnacioBecerra 
For `combo-box`, it looks like the only issue is that when set to `invalid`, the invalid icon appears shifted away from the caret. It should have the same spacing as the `warn` version.

**Current build**
Screen Shot 2023-04-21 at 4 03 51 PM

**Comment by IgnacioBecerra** [comment]
@RichKummer Addressed the warning state on inline dropdown! As for the combobox issue, I couldn't seem to replicate it locally, the invalid icon never shifted from its place. Seems like it could've been a Storybook thing?
 
Screenshot 2023-04-25 at 8 01 46 AM

Screenshot 2023-04-25 at 8 02 14 AM

I also went ahead and added the `With layer` stories as well. 👍

**Review by RichKummer — APPROVED** [review]
The `combo-box` invalid issue may have been Storybook, but it looks good now!

Thanks @IgnacioBecerra !

**Pull request merged**